### PR TITLE
fix(es/minifier): Preserve `cooked` while compressing template literals

### LINKED ISSUE #7770: Some parts of template literals are copied with `minify()`
### Describe the bug

- https://github.com/swc-project/swc/issues/7769

I think this is related to `cm: Lrc<SourceMap>` in the ES codegen

### Input code

```typescript
(self["webpackChunkabc_orchestra"] = self["webpackChunkabc_orchestra"] || []).push([[118],{

    /***/ "./node_modules/path/to/abc.js":
    /***/ (function(__unused_webpack_module, exports, __webpack_require__) {
    
    "use strict";
    
    Object.defineProperty(exports, "__esModule", ({ value: true }));
    exports.MainCSS = exports.sHeight = void 0;
    exports.sHeight = "asdasd";
    const sWidth = "asdasd";
    const sAsideWidth = "weqweqw";
    const sHeight = "asdsads";
    function absolute(abeetew = {}) {
        return `
        abc: cde;
        def: ${abeetew.def || 0};
        qwert: ${abeetew.qwert || 0};
        asda: ${abeetew.asda || 0};
        left: ${abeetew.abc || 0};
    `;
    }
    function flex() {
        return `
        abc: ab;
        cdef: de;
        ghn: fg;
        sadasn: hi;
    `;
    }
    exports.MainCSS = `
    .ThisshouldOnlyBeonTop {
        ${absolute()}
        asda-asdasda: asdad-asdsa;
    }
    .abcBlablaOneQSR {
        line-height: ${exports.sHeight};
        text-shadow: none;
    }
    .abcBlablaOne .asdsad {
        ${flex()}
        ${absolute()}
        
        asdasd-adsdas: asdasd-asdas;
        xzczxc: zxczxcc;
    }
    .asdasd {
        ${absolute({ abc: sAsideWidth + "!important", def: exports.sHeight + "!important", qwert: "0 !important" })}
        sdsd: auto !important;
        xcxcx: auto !important;
    }
    .aasdasdasd .asdsada {
        we: ${sWidth} !important;
    }
    .asdasdasd:not(.adsadsad) .asdasdasd {
        ds: ${parseFloat(exports.sHeight) + parseFloat(sHeight)}rem !important;
    }
    `;
    //# sourceMappingURL=abc.js.map
    
    /***/ })
    
    
    }]);
```

### Config

_No response_

### Playground link

_No response_

### Expected behavior

It should not be copied

### Actual behavior

_No response_

### Version

1.3.74

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

**Related issue:**

 - Closes #7770

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_minifier

File: crates/swc_ecma_minifier/tests/fixture/issues/7770/input.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+(self["webpackChunkabc_orchestra"] = self["webpackChunkabc_orchestra"] || []).push([
```
Comment (jridgewell) [inline crates/swc_ecma_minifier/tests/fixture/issues/7770/input.js:1]: I was able to reduce this down to 
```
const sWidth = "asdasd";
function absolute() {
    return `
    `;
}
function flex() {
    return `
    `;
}
exports.MainCSS = `
.ThisshouldOnlyBeonTop {
    ${absolute()}
}
.abcBlablaOne .asdsad {
    ${flex()}
}
.aasdasdasd .asdsada {
    we: ${sWidth} !important;
}
`;
```

**Review by jridgewell — CHANGES_REQUESTED** [review]
I think this algorithm is still faulty. We never reinitialize `cur_cooked` to a `Some(_)`, so we'll only push cooked values for the first quasi? What about all the other quasis?

I think the better algorithm here is to just peek the last item in `quasis`, and mutate the value directly when we find a string literal.

**Comment by kdy1** [comment]
I added logic for setting `cur_cooked` to `None` to ensure that we only set `cooked` of the new template literal strings if all elements have `cooked` set.

File: crates/swc_ecma_minifier/src/compress/pure/strings.rs [hunk 2]
```diff
@@ -258,13 +264,15 @@ impl Pure<'_> {
...
+                        }
                     }
                     _ => {
                         quasis.push(TplElement {
                             span: DUMMY_SP,
                             tail: true,
-                            cooked: Some(Atom::from(&*cur_cooked)),
+                            cooked: cur_cooked.take().map(From::from),
                             raw: take(&mut cur_raw).into(),
                         });
```
Comment (jridgewell) [inline crates/swc_ecma_minifier/src/compress/pure/strings.rs:277]: We need to reinitialize `cooked = Some(String::new())` here.

**Review by jridgewell — CHANGES_REQUESTED** [review]
My point is that that the second quasi in the following situations will not have a `cooked` value, when it should:

```js
foo`
  \x<-invalid, so no cooked
  ${'this breaks splits the quasis'}
  this should have a cooked
`;

foo`
  \x<-invalid, so no cooked
  ${'this breaks splits the quasis'}
  this should have a cooked
  ${'this breaks splits the quasis'}
  this should also have a cooked
`;
```

File: crates/swc_ecma_minifier/src/compress/pure/strings.rs (hunk 2 above)
Comment (kdy1) [reply crates/swc_ecma_minifier/src/compress/pure/strings.rs:277]: Sorry! I missed the point. And thanks!

**Review by jridgewell — APPROVED** [review]

**Pull request merged**